integrated_channels/canvas/client.py

```diff
@@ -502,17 +502,27 @@ def _extract_integration_id(self, data):
         """
         if not data:
             raise ClientError("No data to transmit.", HTTPStatus.NOT_FOUND.value)
+
         try:
-            integration_id = json.loads(
-                data.decode("utf-8")
-            )['course']['integration_id']
-        except KeyError as error:
+            decoded_payload = data.decode("utf-8")
```

Comment:
separted into new try block because if the data itself is bad, we can't really log the information better in the next block